File: 프로그래머스/lv2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.java
class Solution {
    public String solution(int n, int t, int m, int p) {
        String answer = "";
        String all = "";

        for(int i=0; i<t*m; i++){
            all += Integer.toString(i,n);
        }

        if(m==p){
            p=0;
        }
        for(int i=0; i<all.length(); i++){
            
            if(answer.length() == t){
                break;
            }
            if((i+1)%m == p){
                answer += all.charAt(i);
            }
        }
        return answer.toUpperCase();
    }
}

File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.js
function solution(n, t, m, p) {
    let answer = '';
    const max_len = t * m;
    
    const arr = Array(max_len).fill(0).map((_, i) => i.toString(n).toUpperCase()).join('');
    
    for (let i = 0; i < t; i++) {
        const index = (i * m) + p - 1;
        answer += arr[index];
    }
    
    return answer;
}